spec/formatting.md

```diff
@@ -735,7 +735,7 @@ each _text_ and _placeholder_ part of the selected _pattern_ is resolved and for
 
 Resolved values cannot always be formatted by a given implementation.
 When such an error occurs during _formatting_,
-an implementation SHOULD emit a _Formatting Error_ and produce a
+an implementation SHOULD emit a _Message Function Error_ and produce a
```

Comment:
We'll be able to test for the more general error class as well. @mradbourne or I will file a follow-up PR to his current one to allow for that.

Reply:
Agreed that we should create a test for it. My comment, though, is that we should nail this one down somewhat. Or do you disagree?

Reply:
I disagree. I think leaving it up to the function implementer to pick between the options will provide the best user experience, and it allows for lazy implementations to e.g. only check operands and options until they're absolutely required, and to emit Bad Operand or Bad Option errors at any time.

Reply:
This is a very specific bit of guidance: if there is a fully resolved value _and_ the function determines it cannot be formatted by that function during the formatting process, isn't that the definition of a Bad Operand? Plus, this is `SHOULD` not `MUST`, so an implementation is still permitted to emit something else if it has good reason. 

If I'm an implementer and I'm reading the spec, I'm going to be looking for `MUSTard` that I am required to implement (and the tests that go with it). If we intend to be wiggly, we need to make it clear. In that case, it would look more like:

```suggestion
Resolved values cannot always be formatted by a given implementation.
When a _function_ determines that it does not support the resolve value of its _operand_
its implementation SHOULD emit an informative error, such as _Bad Operand_,
and produce a
```

Reply:
I've updated the text here to now say:
> When such an error occurs during _formatting_,
> an implementation SHOULD emit an appropriate _Message Function Error_ and produce ...

which is in line with @aphillips's suggestion in https://github.com/unicode-org/message-format-wg/pull/774#discussion_r1576395507.